docs/DeveloperGuide.md

```diff
@@ -270,51 +270,189 @@ _{Explain here how the data archiving feature will be implemented}_
+| `* *`    | helpful SoC professor                  | put tags on my student's FYP project so that I can let him/her know whether they are “Ahead of time”, “Up to task” or “Behind schedule” | they can be informed of their progress relative to the timeline [coming soon] |
+| `* *`    | non-confrontational (i.e. shy) student | push certain tags (e.g. “Assistance required!”, “Unable to complete :(”) to the application                                             | gather my professor’s attention without direct confrontation [coming soon]    |
+| `*`      | fun-seeking professor/FYP student      | end “stickers” (similar to Telegram stickers) or emoji so that the app                                                                  | is more appealing to use [coming soon]                                        |
+| `*`      | FYP student                            | update my availability slots on the app                                                                                                 | my professor knows when I will be available to meet [coming soon]             |
+| `*`      | well-versed SoC professor              | sort my student’s FYP based on what specialisation they are taking                                                                      | I can better organise the FYPs [coming soon]                                  |
+| `*`      | very meticulous SoC professor          | see student's FYP sorted by research area first, followed by alphabetical order                                                         | I can gain quick easy access to the details required [coming soon]            |
 
 *{More to be added}*
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `JerylFypManager` and the **Actor** is the `Professor`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: UC01 - Adding FYP**
+
+**Preconditions: Professor knows the details of FYP to be added**
+
+**MSS**
+
+1.  Professor requests to add a student FYP into the list
+2.  JerylFypManager adds the FYP details 
+
+    Use case ends
+
+**Extensions**
+
+* 1a. The request given is invalid 
+  * 1a1. JerylFypManager shows an error message
+  * 1a2. Professor enters a new request
+  
+  Steps 1a1-1a2 are repeated until the request entered is correct
+
+  Use case resumes from step 2
+
+
+**Use case: UC02 - Searching Keyword**
+
+**Preconditions: Professor can recall part of the project name**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  Professor wants to track the status of certain project(s)
+2.  Professor searches the project(s) using a specified keyword
+3.  JerylFypManager displays the search results with a brief summary for each project
 
-    Use case ends.
```

Comment:
Should we have a full stop at the line "Use case ends"?

Reply:
Encouraged to. I don't see why not.